wnfs/src/private/file.rs

```diff
@@ -921,35 +971,44 @@ impl PrivateForestContent {
     /// Reads a number of bytes starting from a given offset.
     pub async fn read_at<'a>(
         &'a self,
-        offset: usize,
-        len: usize,
+        byte_offset: u64,
+        len_limit: Option<usize>,
         forest: &'a impl PrivateForest,
         store: &'a impl BlockStore,
     ) -> Result<Vec<u8>> {
-        let block_content_size = MAX_BLOCK_CONTENT_SIZE;
-        let chunk_size_upper_bound = (self.get_size_upper_bound() - offset).min(len);
+        let block_content_size = MAX_BLOCK_CONTENT_SIZE as u64;
+        let mut chunk_size_upper_bound = self.get_size_upper_bound() - byte_offset as usize;
+
+        if len_limit.is_some() {
+            chunk_size_upper_bound = chunk_size_upper_bound.min(len_limit.unwrap());
+        }
```

Comment:
```suggestion
        if let Some(len_limit) = len_limit {
            chunk_size_upper_bound = chunk_size_upper_bound.min(len_limit);
        }
```
This gets rid of a call to a possibly panicking `.unwrap()` :v: